eventstore/test_views.py

```diff
@@ -387,6 +395,284 @@ def test_postbirth_other_passport_origin(self):
+        """
+        response = self.client.post(self.url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_message_authentication_required(self):
+        """
+        The user needs the `add _message` permission
+        in order to make the request
+
+        """
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+        data = {"random": "data"}
+        response = self.client.post(
+            self.url,
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_message_authentication_errors(self):
+        """
+        We expect to get auth errors when no user token is added to the querystring,
+        but when added the request is made
+        """
+
+        user = get_user_model().objects.create_user("test")
+        user.user_permissions.add(Permission.objects.get(codename="add_message"))
+        data = {"random": "data"}
+
+        response = self.client.post(
+            self.url,
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+        )
+
+        token = Token.objects.create(user=user)
+        url = "{}?token={}".format(reverse("messages-list"), str(token.key))
+        response = self.client.post(
+            url,
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+            HTTP_X_TURN_HOOK_SUBSCRIPTION="whatsapp",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_incorrect_header_request(self):
+        """
+        If the header doesn't contain a value that we recognise,
+        we should return a 400 Bad Request error,
+        explaining that we only accept whatsapp and turn webhook types.
+        """
+        user = get_user_model().objects.create_user("test")
+        user.user_permissions.add(Permission.objects.get(codename="add_message"))
+        self.client.force_authenticate(user)
+        data = {
+            "statuses": [
+                {
+                    "id": "ABGGFlA5FpafAgo6tHcNmNjXmuSf",
+                    "recipient_id": "16315555555",
+                    "status": "read",
+                    "timestamp": "1518694700",
+                    "random": "data",
+                }
+            ]
+        }
+        response = self.client.post(
+            self.url,
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+            HTTP_X_TURN_HOOK_SUBSCRIPTION="other",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
We're not checking that we're explaining why we're returning the error in the response body, we should probably do that